### PR TITLE
Microsoft Teams - add authorization header signature verification

### PR DESCRIPTION
**Pull request by Itay4**
## Status
Ready

## Description
Added auth header signature verification

## Required version of Demisto
5.-

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Code Review

## Dependencies

- [x] MicrosoftTeamsAsk

### CODE REVIEW

**Comment by Itay4** [comment]
@kirbles19 please review the changelog

**Comment by kirbles19** [comment]
@Itay4 Done. Check for technical accuracy.

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 1]
```diff
@@ -537,46 +538,78 @@ def validate_auth_header(headers: dict) -> bool:
...
+    for key in keys:
+        if key.get('kid') == key_id:
+            key_object = key
+            break
+
+    if not key_object:
+        # Didn't find requested key in cache, getting new keys
+        try:
+            open_id_url: str = 'https://login.botframework.com/v1/.well-known/openidconfiguration'
+            response: dict = requests.get(open_id_url).json()
```
Comment (glicht) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]: What about proxy and trust ssl?

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 2]
```diff
@@ -537,46 +538,78 @@ def validate_auth_header(headers: dict) -> bool:
...
+            jwks_uri: str = response.get('jwks_uri', '')
+            keys_response: dict = requests.get(jwks_uri).json()
+            keys = keys_response.get('keys', [])
+            open_id_metadata['keys'] = keys
+        except ValueError:
```
Comment (glicht) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]: Might want to catch also other exceptions. Pretty sure if there are network issues we will see other exceptions.

**Pull request merged**